### DISCUSSION TITLE
How to locally debug? ?debug=true&noredir=true&debugManifestsFile=https://localhost:4321/temp/manifests.js

### DISCUSSION THREAD

**Discussion by Fringie** [microsoft-search/pnp-modern-search#2559] [Q&A] [answered]
Hello,

I need to extend PNP Modern Search and create a new search results layout template. To do this, I need to debug locally, typically for SPFx I've used ?debug=true&noredir=true&debugManifestsFile=https://localhost:4321/temp/manifests.js

Can someone advise how I can debug Modern search custom libraries?

I apologise if I haven't followed the proper GitHub etiquette; I'm a GitHub noobie.

Thanks,
Fringie

**Comment by FranckyC** [comment, marked as answer]
Hi @Fringie, debugging custom libraries is the same as any other SPFx solution. The only thing is you have to first bundle and package the solution (without --ship flag) and upload the .sppkg in the site collection or tenant app catalog. Then run `gulp serve` and open the `https://<your_tenant>.sharepoint.com/sites/your_site/_layouts/15/workbench.aspx` page. If you are using the demo solution, the library should be loaded by default by the main WebPart. Otherwise, make sure you put the correct manifest GUID:

![image](https://user-images.githubusercontent.com/5659122/203124412-3c5f6018-ae3d-4fc0-97c0-2a784eba1bbe.png)

**Reply by Fringie** [reply]
Thank you so much; this worked for me - you saved me a lot of time and headache.
